docs/docs-beta/docs/guides/build/project-structure/creating-a-component.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Creating a New Component Type"
+sidebar_position: 100
+---
+
+:::note
+Refer to the project structuring guide to learn how to create a components-compatible project.
+:::
+
+The `dagster-components` system makes it easy to create new component types that can be reused across your project.
+
+In most cases, component types map to a specific technology. For example, you might have a `DockerScriptComponent` that executes a script in a Docker container, or a `SnowflakeQueryComponent` that runs a query on Snowflake.
+
+## Scaffolding a new component type
+
+For this example, we'll write a lightweight component that executes a shell command.
+
+First, we use the `dg` command-line utility to scaffold a new component type:
+
+```bash
+dg component-type generate shell_command
+```
+
+This will add a new file to your project in the `lib` directory:
+
+<CodeExample filePath="guides/components/shell-command/empty.py" language="python" />
```

Comment:
![Screenshot 2025-01-15 at 9.41.45 AM.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/wS38mdcdU6aAeJobBdry/44192491-1074-4aef-bc8f-f3cd83bd5acb.png)

I'm seeing this locally. Might be a skill issue.